components/server/src/workspace/workspace-deletion-service.ts

```diff
@@ -161,8 +175,13 @@ export class WorkspaceDeletionService {
                     this.config.installationShortname,
                 );
                 const req = new DeleteVolumeSnapshotRequest();
-                req.setId(vs.id);
-                req.setVolumeHandle(vs.volumeHandle);
+                req.setId(vs.vs.id);
+                req.setVolumeHandle(vs.vs.volumeHandle);
+                let type: WorkspaceType = WorkspaceType.REGULAR;
+                if (vs.wsType === "prebuild") {
```

Comment:
I'd be more explicit here and check other types also. We want to prevent this failing when we introduce a new type and it just falls back to Regular here.

Reply:
We only really care here if it is a prebuild or not, as those are treated differently. So if we introduce some new type, it should not matter in this code, unless that new type will have some special volume snapshot handling (and in that case we probably will need to  udpate this code anyway)

Reply:
While that's true right now, the deletion service is simply a client to the API. And the API defines multiple workspace types. We should be defensive in all cases (as it's cheap to do so). 

To put it differently, if I'm gonna experiment and introduce a new WS type, I'd rather see errors in APIs for the type I've introduced then have it do something silently and then go through the pain of tracking it down.

But it's up to you, just suggesting how you can improve resiliency here.

Reply:
@easyCZ PTAL